packages/typespec-client-generator-core/src/types.ts

```diff
@@ -855,7 +858,14 @@ export function getClientTypeWithDiagnostics(
     case "Model":
       retval = diagnostics.pipe(getSdkArrayOrDictWithDiagnostics(context, type, operation));
       if (retval === undefined) {
-        retval = diagnostics.pipe(getSdkModelWithDiagnostics(context, type, operation));
+        const httpPart = getHttpPart(context.program, type);
```

Comment:
For `HttpPart<T>`, TCGC need `getHttPart` to get original `T` wrapped by `HttpPart`.